.github/workflows/images-legacy.yaml

```diff
@@ -168,11 +168,85 @@ jobs:
           path: image-digest
           retention-days: 1
 
+  # we need to build cilium-test separately because we run apt while building the image which requires qemu setup in order to avoid x86/arm64 binaries mixups
+  build-and-push-cilium-test:
+    if: ${{ github.repository == 'cilium/cilium' }}
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        include:
+          - name: cilium-test
+            dockerfile: ./images/cilium-test/Dockerfile
+
+    steps:
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@2a4b53665e15ce7d7049afb11ff1f70ff1610609
+
+      - name: Login to DockerHub to avoid rate limit
+        uses: docker/login-action@f3364599c6aa293cdc2b8391b1b56d0c30e45c8a
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME_CI }}
+          password: ${{ secrets.DOCKER_HUB_PASSWORD_CI }}
+
```

Comment:
This is not required. (I know you are simply copy pasting but it seems like GH is not affected by docker pull rate limits)

Reply:
I've heard unconfirmed rumours of seeing this in github actions output, not sure if related to this PR or not at this point since I don't have a link to the failure:

```
toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
make[1]: *** [Makefile:16: builder-image] Error 123
```